include/mbedtls/ssl.h

```diff
@@ -811,12 +820,10 @@ typedef struct mbedtls_ssl_flight_item mbedtls_ssl_flight_item;
 
 #if defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(MBEDTLS_SSL_SESSION_TICKETS)
 
-typedef enum
-{
-    MBEDTLS_SSL_TICKET_ALLOW_EARLY_DATA = 1,
-    MBEDTLS_SSL_TICKET_ALLOW_DHE_RESUMPTION = 2,
-    MBEDTLS_SSL_TICKET_ALLOW_PSK_RESUMPTION = 4,
-} mbedtls_ssl_ticket_flags;
+typedef uint8_t mbedtls_ssl_tls13_ticket_flags;
+#define MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_PSK_RESUMPTION              ( 1u << 0 )
+#define MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_PSK_EPHEMERAL_RESUMPTION    ( 1u << 2 )
+#define MBEDTLS_SSL_TLS1_3_TICKET_HAS_EARLY_DATA_INDACTION          ( 1u << 3 )
```

Comment:
Otherwise I think that `MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_EARLY_DATA` (as it was) is better (sorry @yuhaoth about that). We can see it as a property or usability of the ticket: the ticket allows early data or not, allows PSK resumption or not ... The presence of an early data indication extension is the EncryptedExtension message is the way for the server to convey the information that the ticket can be used to send early data. And that way it is well aligned with the naming for the PSK(_EPHEMERAL) resumption capability of the ticket.

Reply:
I am not meaning the presence in EncryptedExtension and ClientHello. I mean it is presence of NewSessionTicket. And I think I might understand your meaning.  
On server side,  `allow_early_data` is the status of last session,  if last session or resumption session does not allow early data, server should reject early data, right?

Reply:
> I am not meaning the presence in EncryptedExtension and ClientHello. I mean it is presence of NewSessionTicket.

I meant NewSessionTicket message not EncryptedExtension in my previous comment (fixed).

Reply:
Understand.